Sources/SwiftFrameCore/Workers/ImageWriter.swift

```diff
@@ -0,0 +1,107 @@
+import AppKit
+import Foundation
+
+public final class ImageWriter {
+
+    // MARK: - Exporting
+
+    static func finish(
+        context: CGContext,
+        with outputPaths: [FileURL],
+        sliceSize: CGSize,
+        outputWholeImage: Bool,
+        locale: String,
+        suffix: String,
+        format: FileFormat,
+        completion: @escaping () throws -> Void) throws
+    {
+        guard let image = context.makeImage() else {
+            throw NSError(description: "Could not render output image")
+        }
+        let slices = try sliceImage(image, with: sliceSize)
+
+        let workGroup = DispatchGroup()
+
+        // Writing images asynchronously gave a big performance boost, what a surprise
+        // Also, since we checked beforehand if the directory is writable, we can safely put of the rendering work to a different queue
+        workGroup.enter()
+        DispatchQueue.global().ky_asyncOrExit {
+            try ImageWriter.write(images: slices, to: outputPaths, locale: locale, suffix: suffix, format: format)
+            workGroup.leave()
+        }
+
+        if outputWholeImage {
+            workGroup.enter()
+            DispatchQueue.global().ky_asyncOrExit {
+                try outputPaths.forEach {
+                    try ImageWriter.write(image, to: $0.absoluteURL.appendingPathComponent(locale), fileName: "\(locale)-\(suffix)-big", format: format)
+                }
+                workGroup.leave()
+            }
+        }
+
+        workGroup.wait()
+        workGroup.ky_notifyOrExit(queue: .global()) {
```

Comment:
Does this actually require `notify`? Doesn't the `wait` on the line before guarantee that the group is empty?